pkg/archive/archive.go

```diff
@@ -25,12 +25,17 @@ import (
 )
 
 type (
-	Archive         io.ReadCloser
-	ArchiveReader   io.Reader
-	Compression     int
+	// Archive is a type of io.ReadCloser which has two interfaces Read and Closer.
+	Archive io.ReadCloser
+	// ArchiveReader is a type of io.Reader.
+	ArchiveReader io.Reader
```

Comment:
Naively I would say ``Reader`` :sweat_smile:, that way it does ``archive.Reader``.